.github/workflows/krel-release-notes-validate.yaml

```diff
@@ -0,0 +1,129 @@
+# Copyright 2024 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Yaml Lint Release Notes Pull Request
+on:
+  pull_request:
+    paths:
+      - releases/**/release-notes/**.yaml
+  # Allow manual triggering
+  workflow_dispatch: { }
+
+concurrency:
+  group: ${{ github.ref }}-${{ github.workflow }}
+  cancel-in-progress: true
+jobs:
+  krel_release_notes_validate_action:
+    name: Validate release notes with krel
+    runs-on: ubuntu-latest
+    if: ${{ !github.event.pull_request.draft }}
+    steps:
+      - name: Check out code
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+        with:
+          # we need to fetch the full history in order to check changes across all commits on the branch
+          fetch-depth: '0'
+      - uses: actions/setup-go@41dfa10bad2bb2ae585af6ee5bb4d7d973ad74ed # v5.1.0
+        with:
+          go-version: '1.23'
+          check-latest: true
+      - uses: sigstore/cosign-installer@dc72c7d5c4d10cd6bcb8cf6e3fd625a9e5e537da # v3.7.0
+        with:
+          use-sudo: false
+      - id: install-krel
+        shell: bash
+        run: |
+          #!/bin/bash
+          set -euo pipefail
+          
+          KREL_VERSION=v0.17.12
```

Comment:
I've pinned this to `v0.17.12`, if and when we do a new release of `krel`, we'd need to update this. (I think there may be some automated job to do that, or we can add this to the handbook (for the lead to update it). 

Alternatively, do we want to fetch the latest release?

Reply:
Hm, currently the handbook suggests the docs team uses the lastest krel, so maybe it's fine just fetching the latest release? https://github.com/kubernetes/sig-release/blob/ff03da9043cdd4cfd63f4c06c901909a744f2daa/release-team/role-handbooks/docs/README.md#tools

Reply:
Done, I changed to latest.